pkg/certificate/controller.go

```diff
@@ -44,7 +78,8 @@ func (m *Manager) add(mgr manager.Manager, r reconcile.Reconciler) error {
 			return isWebhookConfigOrAnnotatedResource(createEvent.Meta)
 		},
 		DeleteFunc: func(deleteEvent event.DeleteEvent) bool {
-			return isWebhookConfigOrAnnotatedResource(deleteEvent.Meta)
+			_, isASecret := deleteEvent.Object.(*corev1.Secret)
+			return isASecret && isAnnotatedResource(deleteEvent.Meta) && !isACascadeDelete(deleteEvent)
```

Comment:
This controller watches webhook configurations too. We don't care about those on deletion? That seems to be a change compared to the original code.